File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<long long, int>mp;
        int n = nums.size();
        for(auto v:nums) mp[v]++;
        int i=0;
        int result = 0;
        unordered_map<long long,int>ans;
        while(i<n){
            long long j=nums[i];
            int ct = 0;
            if(ans[j]!=0) {
                i++;
                continue; }
            while(mp[j]){
                j++;
                cout<<"siam"<<endl;
                
                ct++;
                ans[j]=ct;
            }
            result=max(result,ct);
            ans[nums[i]]=ct;
            i++;
        }
        return result;
        
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.java
class Solution {
    public int maxProduct(int[] nums) {
       int n=nums.length;
       boolean[] isComplete=new boolean[n+1];
       int[] dpmax = new int[n+1];
       int[] dpmin = new int[n+1];
       dpmax[0]=nums[0];
       dpmin[0]=nums[0];
       int ans = nums[0];
       for(int i=1;i<n;i++){
        int a = dpmax[i-1]*nums[i];
        int b= dpmin[i-1]*nums[i];
        int c = nums[i];
        
        dpmax[i]=Math.max(a,Math.max(b,c));
        dpmin[i]=Math.min(a,Math.min(b,c));
        ans = Math.max(dpmax[i],ans);
       }
       return ans;

               
    }
   
}
